src/primitive/general_matrix_gate.jl

```diff
@@ -33,6 +33,11 @@ Create a [`GeneralMatrixBlock`](@ref) with a matrix `m`.
 julia> matblock(ComplexF64[0 1;1 0])
 matblock(...)
 ```
+
+!!!warn
+    
+    Instead of converting it to the default data type `ComplexF64`,
+    this will return its contained matrix.
```

Comment:
this is not proper...

Reply:
you don't convert it here.

Reply:
I updated the docstring, explicitly refer to `mat`.